### ISSUE TITLE
proc macro not expanded: Proc macros are disabled

### ISSUE THREAD

**Issue by christopher-wilke** [simrat39/rust-tools.nvim#389]
I see a warning when running `[tokio::main]`:

`proc macro "main" not expanded: Proc macros are disabled`.

Is there any way to disable this warning?

**Comment by 0xDmtri** [comment]
yeah same, this happens if i go to definition and it opens some other dep of the project, it starts building all the stuff and then I get this everywhere too. Super annpying

**Comment by qdrs** [comment]
Yes I have this exact same problem, the macro expansion still works though even after it mentions this error...

**Comment by 52funny** [comment]
I also have this problem. I use the original lsp-config without using rust-tools to solve this problem well, and it is faster than rust-tools.

**Comment by 0xDmtri** [comment]
> I also have this problem. I use the original lsp-config without using rust-tools to solve this problem well, and it is faster than rust-tools.

Do u have inlay hinting there too? If yes, would you mind sharing this part of your config please! Cheers!

**Comment by 52funny** [comment]
> > I also have this problem. I use the original lsp-config without using rust-tools to solve this problem well, and it is faster than rust-tools.
> 
> Do u have inlay hinting there too? If yes, would you mind sharing this part of your config please! Cheers!

You can refer to this：[reddit](https://www.reddit.com/r/neovim/comments/14e41rb/today_on_nightly_native_lsp_inlay_hint_support/)

This post says that Native LSP Inlay Hint is already supported in the Nightly version of neovim.

**Comment by simrat39** [comment]
Is there any way I can reproduce this?

**Comment by 0xDmtri** [comment]
> Is there any way I can reproduce this?

yes ser, all u need to do is to open some rust project, for example, something that uses tokio::spawn(), and then do "go to definition" on "spawn()" method or any other method u r using from an external crate. Rust analyzer will start fetching everything and then it will produce this warning. The warning will stay until u restart the lsp and u will see this warning even in the original code from where you jumped from.

**Comment by aqrln** [comment]
I'm not sure about the exact steps to reproduce but it happens pretty often to me. Any proc macro can trigger this, not necessarily `tokio::main`. `LspRestart` doesn't always help when that happens, only restarting neovim reliably does.
